tests/integration/data_processing/test_data_processor.py

```diff
@@ -376,3 +379,33 @@ def test_categorical_column_with_numbers(self):
 
         # Assert
         assert isinstance(dp._hyper_transformer.field_transformers['category_col'], UniformEncoder)
+
+    def test_update_transformers_id_generator(self):
+        """ Test that updating to transformer to id generator is valid"""
```

Comment:
sorry, for this I meant can we test from the synthesizer level like in the issue. Set up a synthesizer and attempt to assign it the transformer and see if everything works end to end